com.hazelcast.jet.core.JobTest.when_trackedJobCancels_then_jobCompletes

https://hazelcast-l337.ci.cloudbees.com/job/jet-pr-builder/com.hazelcast.jet$hazelcast-jet-core/3212/testReport/junit/com.hazelcast.jet.core/JobTest/when_trackedJobCancels_then_jobCompletes/

```
Error Message
com.hazelcast.jet.impl.exception.JobTerminateRequestedException: CANCEL
Stacktrace
java.util.concurrent.ExecutionException: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: CANCEL
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at com.hazelcast.jet.Job.lambda$join$0(Job.java:92)
	at com.hazelcast.jet.impl.util.Util.uncheckRun(Util.java:111)
	at com.hazelcast.jet.Job.join(Job.java:92)
	at com.hazelcast.jet.core.JobTest.joinAndExpectCancellation(JobTest.java:631)
	at com.hazelcast.jet.core.JobTest.when_trackedJobCancels_then_jobCompletes(JobTest.java:246)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.jet.impl.exception.JobTerminateRequestedException: CANCEL
	at com.hazelcast.jet.impl.MasterContext.getResult(MasterContext.java:635)
	at com.hazelcast.jet.impl.MasterContext.onInitStepCompleted(MasterContext.java:446)
	at com.hazelcast.jet.impl.MasterContext.lambda$invokeOnParticipants$21(MasterContext.java:847)
	at com.hazelcast.jet.impl.util.Util$2.onResponse(Util.java:158)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$1.run(AbstractInvocationFuture.java:255)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:127)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$1.run(AbstractInvocationFuture.java:250)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
```

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java
@@ -629,11 +629,6 @@ public class MasterContext {
             logger.fine(opName + " of " + jobIdString() + " has failures: " + failures);
         }
 
-        TerminationMode mode = requestedTerminationMode;
-        if (mode == CANCEL) {
-            logger.fine(jobIdString() + " to be cancelled after " + opName);
-            return new JobTerminateRequestedException(mode);
-        }
 
         if (successfulMembers.size() == executionPlanMap.size()) {
             logger.fine(opName + " of " + jobIdString() + " was successful");
@@ -647,7 +642,7 @@ public class MasterContext {
         if (failures.stream().allMatch(e -> e.getValue() instanceof TerminatedWithSnapshotException)) {
             assert opName.equals("Execution") : "opName=" + opName;
             logger.fine(opName + " of " + jobIdString() + " terminated after a terminal snapshot");
-
+            TerminationMode mode = requestedTerminationMode;
             assert mode != null && mode.isWithTerminalSnapshot() : "mode=" + mode;
             return new JobTerminateRequestedException(mode);
         }

